posting/views.py

```diff
@@ -1,22 +1,95 @@
 from django.http            import JsonResponse
 from django.views           import View
+from django.db.models       import Count
 
 from user.models    import User
-from posting.models import Posting
+from posting.models import (
+        Posting,
+        PostingSize,
+        PostingHousing,
+        PostingStyle,
+        PostingSpace,
+        PostingLike,
+        PostingComment,
+        PostingScrap
+)
 
 class PostingView(View):
     def get(self, request):
-        postings        = Posting.objects.all()
+        postings = Posting.objects.all()
+        
+        sort_request = request.GET.get('sort', None)
+        postings = postings.annotate(like_num=Count("postinglike"))
+        postings = postings.annotate(comment_num=Count("comment"))
+        postings = postings.annotate(scrap_num=Count("postingscrap"))
+        sort_prefixes = {
+                "like"      : "-like_num",
+                "popular"   : "-comment_num",
+                "scrap"     : "-scrap_num",
+                "recent"    : "-created_at",
+                "old"       : "created_at"
+                }
+        if sort_request in sort_prefixes:
+            postings = postings.order_by(sort_prefixes[sort_request])
+
+        filter_prefixes = {
+                'size_id'       : 'size_id__in',
+                'housing_id'    : 'housing_id__in',
+                'style_id'      : 'style_id__in',
+                'space_id'      : 'space_id__in'
+                }
+        filter_set = {filter_prefixes.ge(key) : value for (key, value) in dict(request.GET).items() if filter_prefixes.get(key)}
```

Comment:
```suggestion
        filter_set = {
            filter_prefixes.ge(key) : value for (key, value) in dict(request.GET).items() 
            if filter_prefixes.get(key)
        }
```
코드가 길어서 가독성을 위해 적절하게 라인 분리 해주시는 것을 추천드립니다.

Reply:
아하.. 넵!!